pyanaconda/modules/boss/module_manager.py

```diff
@@ -0,0 +1,122 @@
+        self._failed_module_services = []
+        self._addon_module_services = []
+
+    @property
+    def addon_module_services(self):
+        return self._addon_module_services
+
+    @property
+    def expected_module_services(self):
+        expected_modules = []
+        # NOTE: we can just initialize expected_modules with the constant,
+        # bacuse the extend afterwards will then *modify* the constant! :P
+        expected_modules.extend(dbus_constants.ANACONDA_MODULES)
+        expected_modules.extend(self.addon_module_services)
+        return expected_modules
+
+    @property
+    def running_module_services(self):
+        # For our purpose, just this
+        return self._started_module_services
+
+    @property
+    def modules_starting_finished(self):
+        return set(self._started_module_services + self._failed_module_services) == set(self.expected_module_services)
+
+    def _start_service_cb(self, service, returned=None, error=None):
+        """Callback for dbus.StartServiceByName."""
+        if error:
+            log.debug("%s error: %s", service, error)
+            self._failed_module_services.append(service)
+        elif returned:
+            if returned == dbus_constants.DBUS_START_REPLY_SUCCESS:
+                log.debug("%s started successfully, returned: %s)", service, returned)
+                self._started_module_services.append(service)
+            else:
+                log.debug("%s failed to start, returned: %s", service, returned)
+                self._failed_module_services.append(service)
+        else:
+            log.error("%s failed to start without even returning anything", service)
+
+        self.check_modules_started()
+
+    def check_modules_started(self):
+        if self.modules_starting_finished:
+            log.debug("modules starting finished, running: %s failed: %s",
+                      self._started_module_services,
+                      self._failed_module_services)
+            for service in self._started_module_services:
+                module = self._bus.get(service)
+                module.EchoString("Boss told me that all required modules were started: %s or failed: %s." %
+                                  (self._started_module_services, self._failed_module_services))
+            return True
+        else:
+            return False
+
+    def find_addons(self):
+        self._addon_module_services = []
+        names = self._dbus.ListActivatableNames()
+        for name in names:
+            if name.startswith(dbus_constants.DBUS_ADDON_NAMESPACE):
+                self._addon_module_services.append(name)
+
+    def start_modules(self):
+        """Starts anaconda modules (including addons)."""
+        log.debug("starting modules")
+
+        # Check that no modules are running
+        for service in self.expected_module_services:
+            if self._dbus.NameHasOwner(service):
+                log.error("service %s has unexpected owner", service)
```

Comment:
This could be the `check_no_modules_running()` method than you don't need comment above.

Reply:
Yeah, let's move it to a method.